connectivity/mbedtls/CMakeLists.txt

```diff
@@ -101,4 +101,6 @@ target_sources(mbed-mbedtls
 target_compile_definitions(mbed-mbedtls
     INTERFACE
         MBED_CONF_MBEDTLS_PRESENT=1
+        # workaround for https://github.com/ARMmbed/mbedtls/issues/1077
+        MULADDC_CANNOT_USE_R7
```

Comment:
Can we make this conditional on an M0 or M0+ core being used?

Reply:
Let me check why this workaround isn't needed for GCC. But anyway the workaround is probably to set it for M0/M0+.

Reply:
Done. The condition also includes M1 and M23 (e.g. NUMAKER_IOT_M263A), added in my latest push.

As for GCC, it compiles exactly the same assembly but doesn't fail, probably related to some optimisation.